sdk/storage/inc/common/memory_stream.hpp

```diff
@@ -32,7 +30,7 @@ namespace Azure { namespace Storage {
     std::size_t m_offset = 0;
   };
 
-  inline std::unique_ptr<MemoryStream> CreateMemoryStream(const uint8_t* data, std::size_t length)
+  inline std::unique_ptr<MemoryStream> CreateMemoryStream(const uint8_t* data, int64_t length)
```

Comment:
I feel the `length` should be `std::size_t`

Reply:
We're going to use int64_t for all of these because we don't want to deal with conversions between these and file sizes. Also, we want to be specific with the ranges.

Reply:
@JeffreyRichter If we don't deal with these conversions, our users will have to.

For example, for a `MemoryStream`, it's very likely the source is a `std::string` or `std::vector<uint8_t>` or `const char*`. The size for STL container is `std::size_t`, and the users will have to static_cast it before calling our function, because otherwise compiler would give a warning.

Reply:
Isn't it better if the customer does the cast because that way they can know if there is a loss of precision or sign issue and then the customer can accommodate it appropriately.

Reply:
okay